File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerStopped.java
package org.kindone.willingtodo.pomodorotimer;

/**
 * Created by kindone on 2017. 1. 1..
 */

public class TimerStopped extends TimerState {

    public TimerStopped(TimerContext context) {
        super(context);
    }

    @Override
    public boolean start() {
        changeStateToRunningWithInheritedDuration();
        return CHANGED;
    }

    @Override
    public boolean stop() {
        // DO NOTHING
        return UNCHANGED;
    }

    @Override
    public boolean resume() {
        // DO NOTHING
        return UNCHANGED;
    }

    @Override
    public boolean pause() {
        // DO NOTHING
        return UNCHANGED;
    }

    public boolean isRunning() {
        return false;
    }

    public boolean isStopped() { return true; }

    @Override
    public long getRemainingTimeMs() {
        return 0;
    }

    private void changeStateToRunningWithInheritedDuration() {
        changeState(new TimerRunning(mContext, mContext.getDurationMs()));
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerStopped.kt
package org.kindone.willingtodo.pomodorotimer

/**
 * Created by kindone on 2017. 1. 1..
 */

class TimerStopped(context: TimerContext) : TimerState(context) {


    override fun start(): Boolean {
        changeStateToRunningWithInheritedDuration()
        return TimerState.CHANGED
    }

    override fun stop(): Boolean {
        // DO NOTHING
        return TimerState.UNCHANGED
    }

    override fun resume(): Boolean {
        // DO NOTHING
        return TimerState.UNCHANGED
    }

    override fun pause(): Boolean {
        // DO NOTHING
        return TimerState.UNCHANGED
    }

    override fun isRunning(): Boolean {
        return false
    }

    override fun isStopped(): Boolean {
        return true
    }

    override fun getRemainingTimeMs(): Long {
        return 0
    }

    override fun getStartedTimeMs(): Long {
        return -1L
    }

    private fun changeStateToRunningWithInheritedDuration() {
        changeState(TimerRunning(mContext, mContext.getDurationMs()))
    }
}
